prow/tide/tide.go

```diff
@@ -1407,7 +1385,7 @@ func prMergeMethod(c config.Tide, crc *CodeReviewCommon) (types.PullRequestMerge
 	return method, nil
 }
 
-func (c *syncController) mergePRs(sp subpool, prs []CodeReviewCommon) error {
+func (gi *GitHubProvider) mergePRs(sp subpool, prs []CodeReviewCommon, dontUpdateStatus *threadSafePRSet) error {
```

Comment:
Currently all of the status updating logic is GitHub specific and due to the differences between GitHub and Gerrit, it seems to me like it could remain that way. I'm wondering if we shouldn't worry about generalizing the status updater and instead just fold it into `GitHubProvider`.  We could then also remove `dontUpdateStatus` from this interface method and instead store it in `GitHubProvider`. This could simplify things when we go to implement the `GerritProvider`.

Reply:
The current plan is to implement status controller for each provider, and currently the gerrit status controller will do nothing. I agree that fold status controller into GitHubController is a good idea, but not sure that I want to further complicate this current PR. Will evaluate and create a new PR for this purpose if it works

Reply:
gave it s try, and quickly ran into a problem:
- https://github.com/kubernetes/test-infra/blob/55d672fc3490d9fd8a8cfaf9e37fe2a09e33b229/prow/tide/status.go#L281, status controller needs to reference functions from GitHubProvider. It could be resolved by refactoring GitHubProvider so that shared functions are part of another sub struct, but I'm afraid that it will not be very clean or easy to understand.

I'll leave it as is for now before figuring out a more elegant solution